api/spec/controllers/mno_enterprise/jpi/v1/admin/cloud_apps_controller_spec.rb

```diff
@@ -3,102 +3,60 @@
-      api_stub_for(put: "/apps/#{app.id}", response: from_api(app))
+      stub_user(user)
+      stub_api_v2(:get, "/apps/#{app.id}", app)
+      stub_api_v2(:patch, "/apps/#{app.id}", app)
+      sign_in user
     end
 
-    context "admin user" do
-      before do
-        sign_in admin
-      end
-
-      describe "#index" do
+    context 'admin user' do
+      describe '#index' do
         subject { get :index }
-
+        let!(:stub) { stub_api_v2(:get, "/apps", [app], [], { filter: { stack: 'cloud' } }) }
         it_behaves_like 'a jpi v1 admin action'
-
         it 'returns the cloud aplications' do
           subject
-          expect(JSON.parse(response.body)).to eq({"cloud_apps"=>[{"id"=>app.id, "uid"=>app.uid, "name"=>app.name, "api_key"=>app.api_key, "tiny_description"=>app.tiny_description, "description"=>app.description, "metadata_url"=>nil, "details"=>nil, "terms_url"=>app.terms_url}]})
+          expect(JSON.parse(response.body)).to eq({ "cloud_apps" => [{ "id" => app.id, "uid" => app.uid, "name" => app.name, "api_key" => app.api_key, "tiny_description" => app.tiny_description, "description" => app.description, "metadata_url" => nil, "details" => nil, "terms_url" => app.terms_url }] })
+        end
+        it 'does the request' do
+          subject
+          expect(stub).to have_been_requested
         end
       end
 
       describe '#update' do
-        let(:params) { {name: 'CloudApp', terms_url: 'terms.com'} }
+        let(:params) { { name: 'CloudApp', terms_url: 'terms.com' } }
+        before { stub_api_v2(:get, "/apps/#{app.id}", app) }
+        let!(:stub) { stub_api_v2(:patch, "/apps/#{app.id}", app) }
         subject { put :update, id: app.id, cloud_app: params }
-
-        before { allow(MnoEnterprise::App).to receive(:find) { app } }
-
         it_behaves_like 'a jpi v1 admin action'
-
-        it 'assigns the cloud app' do
-          subject
-          expect(assigns[:cloud_app]).to eq(app)
-        end
-
-        it 'updates the cloud app' do
-          expect(app).to receive(:save) { true }
-          subject
-          expect(app.terms_url).to eq(params[:terms_url])
-        end
-
-        it 'only updates authorized fields' do
-          subject
-          expect(app.name).to eq('My App')
-        end
-
-        it 'is successful' do
+        it do
           subject
+          expect(stub).to have_been_requested
           expect(response).to be_success
         end
       end
 
-      describe "#regenerate_api_key" do
+      describe '#regenerate_api_key' do
         subject { put :regenerate_api_key, id: app.id }
-
+        let!(:stub) { stub_api_v2(:patch, "/apps/#{app.id}/regenerate_api_key", app) }
         it_behaves_like 'a jpi v1 admin action'
-
-        it 'regenerates the API key' do
-          expect_any_instance_of(MnoEnterprise::App).to receive(:regenerate_api_key!)
+        it do
           subject
-        end
-      end
-
-      describe "#refresh_metadata" do
```

Comment:
Why has this example been removed?

Reply:
I should have removed the method too. The MnoHub endpoint has not been implemented.